src/common/createplay/CreatePlay.jsx

```diff
@@ -89,86 +138,103 @@ const CreatePlay = () => {
     return (
       <Loader
         title={loadingText || "Loading authentication information"}
-        subtitle="Please wait...."
+        subtitle='Please wait....'
       />
     );
   }
 
-  if (!isAuthenticated) {
+  if (!isAuthenticated && !isEditPlay) {
     window.location = NHOST.AUTH_URL(
-      `http://localhost:${
-        process.env.RAECT_APP_DEV_PORT ?? "3000"
-      }/plays/create`
+      `http://localhost:${process.env.RAECT_APP_DEV_PORT ?? "3000"}/plays/create`
     );
     return null;
+  } else if (!isAuthenticated && isEditPlay) {
+    window.location = NHOST.AUTH_URL(window.location.href, "github");
+    return null;
   } else {
     initializeData();
   }
 
   if (isDataLoading) {
-    return <Loader title={"Loading data"} subtitle="Please wait...." />;
+    <Loader title={"Loading data"} subtitle='Please wait....' />;
+  }
+
+  if (isEditPlay && !creator && !errorMessage) {
+    return (
+      <PageNotFound msg="You don't own this play" details='Only Owner is able to edit their play' />
+    );
+  }
+
+  if (errorMessage) {
+    return <PageNotFound msg='Something Error Occured' details='apologize for the inconvenience' />;
   }
 
   return (
-    <div className="w-full h-full flex flex-col justify-center items-center create-plays-wrapper">
+    <div className='w-full h-full flex flex-col justify-center items-center create-plays-wrapper'>
       <div>
-        <span className="title-primary">
-          <h1>
-            Create <strong>A Play</strong>
-          </h1>
+        <span className='title-primary'>
+          {!isEditPlay ? (
+            <>
+              Create <strong>A Play</strong>
+            </>
+          ) : (
+            <>
+              Edit <strong>Play</strong>
+            </>
+          )}
         </span>
       </div>
-      <div className="w-full h-full max-w-6xl flex bg-white shadow-md rounded mb-6">
-        <div className="flex flex-col flex-1">
-          <div className="h-14 p-8">
-            Welcome <strong>{userData.displayName}</strong>, create your play
-          </div>
-
-          <div className="flex-1 px-10 py-8 overflow-auto">
-            <form>
-              <PlayForm fields={FIELD_TEMPLATE} onChange={data => onChange(data)} />
-            </form>
-          </div>
-          <div className="h-14">
-            <hr />
-            <div className="p-8 h-full flex items-center">
-              <Button size="small" variant="contained" disabled={isFieldsAreInValid()} onClick={() => onSubmit()}>
-                Create the awesome
-              </Button>
-            </div>
+      <div className='w-full h-full max-w-6xl flex bg-white shadow-md rounded mb-6'>
+        <div className='flex flex-col flex-1'>
+          <div className='h-14 p-8'>
+            Welcome <strong>{userData.displayName}</strong>, {!isEditPlay ? "create" : "edit"} your
+            play
           </div>
+          <PlayForm
+            isEditPlay={isEditPlay}
+            fields={FIELD_TEMPLATE}
```

Comment:
Does it make sense to allow editing of all the fields?
like
- What is the usecase of editing github username? Was it just a typo
- What is the usecase of editing language? One needs to translate a JS play to TS? Is it a valuable exercise?
- etc

cc @atapas

Reply:
actually i have disabled these two field when the form is mounted to be edit play. i didnt remove them since the same form component is used to create play and i might mess it up removing it :) . also i guess we wont be sending github_username and issue for edit play query can ensure these two field has no use for edit play other than showing the value.

looking forward for other thoughts.

![Screenshot from 2022-11-08 08-37-48](https://user-images.githubusercontent.com/63530626/200465654-3ebbf8a0-77d2-4896-898e-5664d9d86924.png)